src/libs/actions/Report.ts

```diff
@@ -2690,15 +2688,6 @@ function openReportFromDeepLink(url: string) {
 
                         const state = navigationRef.getRootState();
                         const currentFocusedRoute = findFocusedRoute(state);
-                        const hasCompletedGuidedSetupFlow = hasCompletedGuidedSetupFlowSelector(onboarding);
-
-                        // We need skip deeplinking if the user hasn't completed the guided setup flow.
-                        if (!hasCompletedGuidedSetupFlow) {
-                            Welcome.isOnboardingFlowCompleted({
-                                onNotCompleted: () => OnboardingFlow.startOnboardingFlow(),
-                            });
-                            return;
-                        }
```

Comment:
Are you sure we can fully remove this?

Reply:
I'll revert it for now, as in some flows we might never hit the redirect to onboarding because of this change.

Reply:
I think the logic there is redundant a bit, right? we check if the modal should show and then we call `Welcome.isOnboardingFlowCompleted` which does the same technically, so we could refactor this to only check `hasCompletedGuidedSetupFlowSelector` and run `OnboardingFlow.startOnboardingFlow()` if the modal should show.

but we can leave this for later

Reply:
Oh right, that looks redundant too. I was focusing around the fact that we also navigate the user in `NavigationRoot.tsx`:
```ts
        // If the user haven't completed the flow, we want to always redirect them to the onboarding flow.
        // We also make sure that the user is authenticated.
        if (!NativeModules.HybridAppModule && !hasCompletedGuidedSetupFlow && authenticated && !shouldShowRequire2FAModal) {
            const {adaptedState} = getAdaptedStateFromPath(getOnboardingInitialPath(), linkingConfig.config);
            return adaptedState;
        }
```